File: ALODWebUtility/Common/Message.cs
using System;
using System.Data;
using System.Data.Common;
using ALOD.Data;

namespace ALODWebUtility.Common
{
    [Serializable]
    public class Message
    {
        #region Members

        protected byte _rows;
        private bool _assigned = false;
        private DateTime _endTime = new DateTime();
        private byte _groupId = 0;
        private string _groupName = string.Empty;
        private bool _isAdmin = false;
        private string _message = string.Empty;
        private short _messageId = 0;
        private string _name = string.Empty;
        private bool _popup = false;
        private DateTime _startTime = new DateTime();
        private string _title = string.Empty;

        #endregion

        #region Properties

        public bool Assigned
        {
            get { return _assigned; }
            set { _assigned = value; }
        }

        public DateTime EndTime
        {
            get { return _endTime; }
            set { _endTime = value; }
        }

        public byte GroupId
        {
            get { return _groupId; }
            set { _groupId = value; }
        }

        public string GroupName
        {
            get { return _groupName; }
            set { _groupName = value; }
        }

        public bool IsAdmin
        {
            get { return _isAdmin; }
            set { _isAdmin = value; }
        }

        public string MessageText // Renamed from Message to MessageText to avoid conflict with class name
        {
            get { return _message; }
            set { _message = value; }
        }

        public short MessageId
        {
            get { return _messageId; }
            set { _messageId = value; }
        }

        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        public bool Popup
        {
            get { return _popup; }
            set { _popup = value; }
        }

        public DateTime StartTime
        {
            get { return _startTime; }
            set { _startTime = value; }
        }

        public string Title
        {
            get { return _title; }
            set { _title = value; }
        }

        #endregion

        public void DeleteMessage(short messageId)
        {
            SqlDataStore adapter = new SqlDataStore();
            DbCommand cmd;
            cmd = adapter.GetStoredProcCommand("core_messages_sp_DeleteMessage");
            adapter.AddInParameter(cmd, "@messageId", DbType.Int16, messageId);
            adapter.ExecuteNonQuery(cmd);
        }

        /// <summary>
        /// Returns a Message details row
        /// </summary>
        /// <param name="messageId">The ID of the Message</param>
        /// <returns>A data row containing the results.</returns>
        /// <remarks></remarks>
        public Message RetrieveMessageDetails(int messageId)
        {
            SqlDataStore adapter = new SqlDataStore();
            DbCommand cmd = adapter.GetStoredProcCommand("core_messages_sp_GetMessagesDetails", messageId);
            adapter.ExecuteReader(RetrieveMessageDetailsReader, cmd);
            return this;
        }

        protected void RetrieveMessageDetailsReader(SqlDataStore adapter, IDataReader reader)
        {
            _rows += 1;

            //0-Title, 1-Message, 2-Name, 3-Popup, 4-StartTime,5-EndTime
            //Dim message As New Message
            _title = adapter.GetString(reader, 0);
            _message = adapter.GetString(reader, 1);
            _name = adapter.GetString(reader, 2);
            _popup = adapter.GetBoolean(reader, 3);
            _startTime = DateTime.Parse(adapter.GetDateTime(reader, 4, DateTime.Now).ToShortDateString());
            _endTime = DateTime.Parse(adapter.GetDateTime(reader, 5, DateTime.Now).ToShortDateString());
        }
    }
}

File: ALODWebUtility/Common/Message.vb
Imports ALOD.Data

Namespace Common

    <Serializable()>
    Public Class Message

#Region "Members"

        Protected _rows As Byte
        Private _assigned As Boolean = False
        Private _endTime As New Date
        Private _groupId As Byte = 0
        Private _groupName As String = String.Empty
        Private _isAdmin As Boolean = False
        Private _message As String = String.Empty
        Private _messageId As Int16 = 0
        Private _name As String = String.Empty
        Private _popup As Boolean = False
        Private _startTime As New Date
        Private _title As String = String.Empty

#End Region

#Region "Properties"

        Public Property Assigned() As Boolean
            Get
                Return _assigned
            End Get
            Set(ByVal value As Boolean)
                _assigned = value
            End Set
        End Property

        Public Property EndTime() As Date
            Get
                Return _endTime
            End Get
            Set(ByVal value As Date)
                _endTime = value
            End Set
        End Property

        Public Property GroupId() As Byte
            Get
                Return _groupId
            End Get
            Set(ByVal value As Byte)
                _groupId = value
            End Set
        End Property

        Public Property GroupName() As String
            Get
                Return _groupName
            End Get
            Set(ByVal value As String)
                _groupName = value
            End Set
        End Property

        Public Property IsAdmin() As Boolean
            Get
                Return _isAdmin
            End Get
            Set(ByVal value As Boolean)
                _isAdmin = value
            End Set
        End Property

        Public Property Message() As String
            Get
                Return _message
            End Get
            Set(ByVal value As String)
                _message = value
            End Set
        End Property

        Public Property MessageId() As Int16
            Get
                Return _messageId
            End Get
            Set(ByVal value As Int16)
                _messageId = value
            End Set
        End Property

        Public Property Name() As String
            Get
                Return _name
            End Get
            Set(ByVal value As String)
                _name = value
            End Set
        End Property

        Public Property Popup() As Boolean
            Get
                Return _popup
            End Get
            Set(ByVal value As Boolean)
                _popup = value
            End Set
        End Property

        Public Property StartTime() As Date
            Get
                Return _startTime
            End Get
            Set(ByVal value As Date)
                _startTime = value
            End Set
        End Property

        Public Property Title() As String
            Get
                Return _title
            End Get
            Set(ByVal value As String)
                _title = value
            End Set
        End Property

#End Region

        Public Sub DeleteMessage(ByVal messageId As Int16)
            Dim adapter As New SqlDataStore
            Dim cmd As Data.Common.DbCommand
            cmd = adapter.GetStoredProcCommand("core_messages_sp_DeleteMessage")
            adapter.AddInParameter(cmd, "@messageId", Data.DbType.Int16, messageId)
            adapter.ExecuteNonQuery(cmd)
        End Sub

        ''' <summary>
        ''' Returns a Message details row
        ''' </summary>
        ''' <param name="messageId">The ID of the Message</param>
        ''' <returns>A data row containing the results.</returns>
        ''' <remarks></remarks>
        Public Function RetrieveMessageDetails(ByVal messageId As Integer) As Message
            Dim adapter As New SqlDataStore()
            Dim cmd As System.Data.Common.DbCommand = adapter.GetStoredProcCommand("core_messages_sp_GetMessagesDetails", messageId)
            adapter.ExecuteReader(AddressOf RetrieveMessageDetailsReader, cmd)
            Return Me

        End Function

        Protected Sub RetrieveMessageDetailsReader(ByVal adapter As SqlDataStore, ByVal reader As System.Data.IDataReader)
            _rows += 1

            '0-Title, 1-Message, 2-Name, 3-Popup, 4-StartTime,5-EndTime
            'Dim message As New Message
            _title = adapter.GetString(reader, 0)
            _message = adapter.GetString(reader, 1)
            _name = adapter.GetString(reader, 2)
            _popup = adapter.GetBoolean(reader, 3)
            _startTime = adapter.GetDateTime(reader, 4, Now).ToShortDateString
            _endTime = adapter.GetDateTime(reader, 5, Now).ToShortDateString

        End Sub

    End Class

End Namespace